File: app/src/main/java/com/example/haidermirza/banner/viper/contract/OverFlyingContract.java
package com.example.haidermirza.banner.viper.contract;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.widget.SeekBar;

import com.example.haidermirza.banner.library.BannerLayout;

public interface OverFlyingContract {
    interface OF_View{
        void show_message(String str);
    }
    interface OF_Intrector{
        void attach_presenter(OverFlyingContract.OF_Presenter presenter);
        void initial_dataLoad(RecyclerView recyclerView, SeekBar seekBar, Context context);
    }
    interface OF_Presenter{
        void attach_view(OverFlyingContract.OF_View view);
        void on_banner_clik(String path);
        void on_initial_dataLoad(RecyclerView recyclerView, SeekBar seekBar, Context context);
    }
}

File: app/src/main/java/com/example/haidermirza/banner/viper/contract/OverFlyingContract.kt
package com.example.haidermirza.banner.viper.contract

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.widget.SeekBar

interface OverFlyingContract {
    interface OF_View {
        fun show_message(str: String)
    }

    interface OF_Intrector {
        fun attach_presenter(presenter: OverFlyingContract.OF_Presenter)
        fun initial_dataLoad(recyclerView: RecyclerView, seekBar: SeekBar, context: Context)
    }

    interface OF_Presenter {
        fun attach_view(view: OverFlyingContract.OF_View)
        fun on_banner_clik(path: String)
        fun on_initial_dataLoad(recyclerView: RecyclerView, seekBar: SeekBar, context: Context)
    }
}
